program/src/state/rules.rs

```diff
@@ -249,26 +255,52 @@ impl Rule {
                     (false, self.to_error())
                 }
             }
-            Rule::DerivedKeyMatch { account, field } => {
-                msg!("Validating DerivedKeyMatch");
+            Rule::PDAMatch {
+                program,
+                pda_field,
+                seeds_field,
+            } => {
+                msg!("Validating PDAMatch");
 
-                let seeds = match payload.get_seeds(field) {
+                // Get the PDA from the payload.
+                let account = match payload.get_pubkey(pda_field) {
+                    Some(pubkey) => pubkey,
+                    _ => return (false, RuleSetError::MissingPayloadValue.into()),
+                };
+
+                // Get the derivation seeds from the payload.
+                let seeds = match payload.get_seeds(seeds_field) {
```

Comment:
I think if we fetch everything from the Payload which the user can add to, then this isn't really any more trustless than ProgramOwned.

Reply:
If someone only uses ProgramOwned, the fact that wallet accounts can be re-assigned to be owned by a program makes it easy for there to be workarounds of auth rules.  @febo and I discovered this with SPL transfers.

If someone puts in the wrong seeds then they are transferring to a PDA they wouldn't have control over.  It would be like burning your NFT.  Or if trying to transfer out of a PDA if you put in the wrong seeds the NFT just wouldn't be there.

If they pass in the seeds and program is not stored in the rule, then they could be passing to their own PDA.  But the RuleSet should also be using ProgramOwned for that case.  Maybe we don't want to offer that.  If we require program to be in the Rule I think its safer.

Please review my logic in the above comments and tell me if I'm misunderstanding the situation.